flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/SpillableSubpartition.java

```diff
@@ -146,23 +146,26 @@ public void finish() throws IOException {
 
 	@Override
 	public void release() throws IOException {
+		// view reference accessible outside the lock, but assigned inside the locked scope
 		final ResultSubpartitionView view;
 
 		synchronized (buffers) {
 			if (isReleased) {
 				return;
 			}
 
+			// Release all available buffers
```

Comment:
one more nit: I would drop this comment, since the code is quite obvious. If you do not agree, then extracting it to private method named `relaseBuffers()` would also be better, but that's nit picking so you can ignore this.